File: docs/api/spring/com/shoptourr/api/v1/dto/diary/DiaryDtos.java
package com.shoptourr.api.v1.dto.diary;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

import java.time.Instant;
import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

/** Diary screen. */
public final class DiaryDtos {

    private DiaryDtos() {}

    public record DiaryEntryDto(
            UUID id,
            UUID tripId,
            LocalDate entryDate,
            /** Emoji mood from mock set. */
            String mood,
            String text,
            Instant createdAt,
            Instant updatedAt
    ) {}

    public record CreateDiaryEntryRequest(
            /** Defaults to today if null. */
            LocalDate entryDate,
            @NotBlank @Size(min = 1, max = 8) String mood,
            @NotBlank @Size(min = 1, max = 4000) String text
    ) {}

    public record UpdateDiaryEntryRequest(
            @Size(min = 1, max = 8) String mood,
            @Size(min = 1, max = 4000) String text
    ) {}

    public record DiaryDayGroupDto(
            LocalDate date,
            String labelKey,
            List<DiaryEntryDto> entries
    ) {}

    public record TripDiaryResponse(
            List<DiaryDayGroupDto> days
    ) {}
}

File: shared/src/commonMain/kotlin/com/example/shoptourr/api/diary/DiaryDtos.kt
package com.example.shoptourr.api.diary

data class DiaryEntryDto(
    val id: String,
    val tripId: String,
    val entryDate: String,
    val mood: String,
    val text: String,
    val createdAt: String,
    val updatedAt: String,
)

data class CreateDiaryEntryRequest(
    val entryDate: String? = null,
    val mood: String,
    val text: String,
)

data class UpdateDiaryEntryRequest(
    val mood: String? = null,
    val text: String? = null,
)

data class DiaryDayGroupDto(
    val date: String,
    val labelKey: String? = null,
    val entries: List<DiaryEntryDto>,
)

data class TripDiaryResponse(
    val days: List<DiaryDayGroupDto>,
)
